pyuvdata/uvdata/mir_meta_data.py

```diff
@@ -0,0 +1,3087 @@
+            `return_header_keys=True`, then instead of a boolean array, a list of ints
+            (or tuples of ints if the MetaDataObject has only a pseudo header key)
+            corresponding to the header key values.
+
+        Raises
+        ------
+        ValueError
+            If `select_comp` is not one of the permitted strings, or if `select_field`
+            is not one of the fields within `data_arr`.
+        """
+        if select_field not in self._data.dtype.names:
+            raise MirMetaError(
+                "select_field %s not found in structured array." % select_field
+            )
+
+        # Create a simple dict to match operation keywords to a function.
+        op_dict = {
+            "eq": np.equal,
+            "ne": np.not_equal,
+            "lt": np.less,
+            "le": np.less_equal,
+            "gt": np.greater,
+            "ge": np.greater_equal,
+            "between": lambda val, lims: ((val >= lims[0]) & (val <= lims[1])),
+            "outside": lambda val, lims: ((val < lims[0]) | (val > lims[1])),
+        }
+
+        if isinstance(select_val, (list, set, tuple, str, np.ndarray, np.str_)):
+            op_dict["eq"] = lambda val, comp: np.isin(val, comp)
+            op_dict["ne"] = lambda val, comp: np.isin(val, comp, invert=True)
+
+        op_dict["=="] = op_dict["eq"]
+        op_dict["!="] = op_dict["ne"]
+        op_dict["<"] = op_dict["lt"]
+        op_dict["<="] = op_dict["le"]
+        op_dict[">"] = op_dict["gt"]
+        op_dict[">="] = op_dict["ge"]
+
+        # Make sure the inputs look valid
+        if select_comp not in op_dict:
+            raise ValueError('select_comp must be one of: "%s"' % '", "'.join(op_dict))
+
+        # Evaluate data_arr now
+        data_mask = op_dict[select_comp](self._data[select_field], select_val)
+
+        # Trap a corner-case here (most commonly w/ we-read), where some attributes
+        # are multi-dim arrays rather than singleton values per index position.
+        while data_mask.ndim > 1:
+            data_mask = np.any(data_mask, axis=-1)
+
+        # Apply the mask now if an argument has been supplied for it.
+        if mask is not None:
+            data_mask &= mask
+
+        if return_header_keys and self._header_key is None:
+            return list(
+                zip(*[self._data[key][data_mask] for key in self._pseudo_header_key])
+            )
+        elif return_header_keys:
+            return self._data[self._header_key][data_mask]
+
+        return data_mask
+
+    def _index_query(
+        self,
+        use_mask=None,
+        where=None,
+        and_where_args=True,
+        header_key=None,
+        index=None,
+    ):
+        """
+        Find array index positions where selection criteria are met.
+
+        This is an internal helper function used by several methods of the MirMetaData
+        class, and is not designed for users, but instead is part of the developer API.
```

Comment:
I might not use the term "API" here -- it might lead people to think it's a stable interface, which is usually not guaranteed for methods that start with an underscore.

Reply:
Fixed in a2f46b4b.

Reply:
Fixed in a2f46b4b.